gym/core.py

```diff
@@ -106,6 +87,12 @@ class Env(Generic[ObsType, ActType], metaclass=decorator):
     Note: a default reward range set to :math:`(-\infty,+\infty)` already exists. Set it if you want a narrower range.
     """
 
+    def __init_subclass__(cls) -> None:
+        """Hook used for wrapping render method."""
+        super().__init_subclass__()
+        if "render" in vars(cls):
+            cls.render = _deprecate_mode(vars(cls)["render"])
```

Comment:
Any reason why this can't be `cls.render = _deprecate_mode(cls.render)`? If not, I'd prefer that for simplicity

Reply:
Given `"render" in vars(cls)` is `True`, `vars(cls)["render"]` is the same as `cls.render` in 99.9% cases, except `__getattr__` being overridden.
I want to maximize the consistency between exist-check and access. Ideally I want to write `if "render" in vars(cls): vars(cls).update(render=vars(cls)["render"])`, but `vars(cls)` is read-only.
FWIW, the reason why `try...except` or `hasattr` check + `cls.render` cannot be used is that `cls.render` can come from base classes.

Reply:
I like the alternative you presented here, of course with the added wrapper